packages/bot-skeleton/src/scratch/hooks/block_svg.js

```diff
@@ -67,10 +68,12 @@ Blockly.BlockSvg.prototype.showContextMenu_ = function (e) {
     // Save the current block in a variable for use in closures.
     const block = this;
     const menu_options = [];
+    if (!config.mainBlocks.includes(block.type) && this.isDeletable() && this.isMovable() && !block.isInFlyout) {
```

Comment:
do we need the first check at all, since we checking for the parent block at line 193 and `main` blocks never have parents. Thus we will have the `Detach block` option disabled for main blocks same as `Duplicate` and there will be no need for additional import

Reply:
and for consistency i think its better to use `this` instead of `block`. We are saving `this` to `block` variable to use in callbacks. Please refer to this [discussion](https://groups.google.com/g/blockly/c/ZMq3Lzgd9ZM/m/VnFvcpbVBwAJ)